spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindableProxyFactory.java

```diff
@@ -252,7 +104,7 @@ public synchronized Object invoke(MethodInvocation invocation) throws Throwable
 	@Override
 	public synchronized Object getObject() throws Exception {
 		if (this.proxy == null) {
-			createChannels(this.type);
+			this.channelFactory.createChannels(this.type);
```

Comment:
I believe `createChannels` needs to be called before `getInputs` and `getOutputs` as the createChannels method would have the `inputs` and `outputs` set with appropriate channelBindingNames/MessageChannel. What am I missing?